core/src/main/java/org/infinispan/cache/impl/DecoratedCache.java

```diff
@@ -400,7 +477,7 @@ public boolean isEmpty() {
 
    @Override
    public boolean containsKey(Object key) {
-      return cacheImplementation.containsKey(key, flags);
+      return get(key) != null;
```

Comment:
Actually I find that changing the CacheImpl methods to take the `InvocationContext` makes the `DecoratedCache` a bit nicer. So I will try that out.